torchdata/datapipes/iter/load/fsspec.py

```diff
@@ -87,8 +91,10 @@ def __iter__(self) -> Iterator[str]:
                     else:
                         if is_local:
                             abs_path = os.path.join(path, file_name)
-                        else:
+                        elif not file_name.startswith(path):
                             abs_path = posixpath.join(path, file_name)
+                        else:
+                            abs_path = file_name
```

Comment:
Just to confirm - this never happens if `is_local == True`?

Reply:
Good question. I can't guarantee it since I am not aware of all use cases.Based on my experience on `fsspec`, I only encounter the problem when the input is `s3` url.
For local files, I would trust the test here https://github.com/pytorch/data/blob/main/test/test_fsspec.py#L56-L65. As long as it doesn't break, I think